src/java.base/share/classes/java/net/DatagramPacket.java

```diff
@@ -394,8 +394,9 @@ public synchronized void setData(byte[] buf) {
      * Set the length for this packet. The length of the packet is
      * the number of bytes from the packet's data buffer that will be
      * sent, or the number of bytes of the packet's data buffer that
-     * will be used for receiving data. The length must be lesser or
-     * equal to the offset plus the length of the packet's buffer.
+     * will be used for receiving data. The {@code length} plus the
+     * {@link #getOffset() offset} must be lesser or equal to the
+     * length of the packet's data buffer.
```

Comment:
The new text looks good to me. While you're at it I would suggest to rewrite the code that checks the length with something like:

```
        Preconditions.checkFromIndexSize(offset, length, buf.length,
                Preconditions.outOfBoundsExceptionFormatter(IllegalArgumentException::new));
```

Then figure out if we have a unit test that already checks this, and if not, write one!

Reply:
Hello Daniel, thank you for the review. I've updated the PR to implement the code change you suggested.

There's already `test/jdk/java/net/DatagramPacket/Setters.java` which has `testSetLength` to test this method. I've run the entire `test/jdk/java/net/DatagramPacket` tests with this change and they passed fine. I'll trigger a more comprehensive test run just to make sure nothing unexpected shows up.